### PR TITLE
[BAC-105] multiple k8s versions ci test all

### PR DESCRIPTION
**Pull request by coffeepac**
adds support for testing non-default versions of k8s in kraken-lib

### CODE REVIEW

**Comment by coffeepac** [comment]
This is almost ready for merge.  Need to add some documentation.

**Comment by coffeepac** [comment]
added some documentation.  @guineveresaenger good enough?  We don't have a big doc for what's up with CI in general and that's probably out of scope for this PR.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -89,6 +89,7 @@ Please submit all contributions as pull requests, using this process:
    * To quickly create a topic branch based on the master:
      `git checkout -b fix/master/my_contribution master`
    * Avoid working directly on the master branch
+   * If your change probably affects backwards compatability, please include 'test-all' in the feature branch name.  This will cause our CI to run against all supported versions of kubernetes
```
Comment (guineveresaenger) [inline CONTRIBUTING.md]: May we have an example for exact syntax?
e.g. `git br mypatchfix-test-all`?

Comment (guineveresaenger) [inline CONTRIBUTING.md]: May we have an example for exact syntax?
e.g. `git br mypatchfix-test-all?`

File: Documentation/UPGRADING_KUBERNETES.md [hunk 2]
```diff
@@ -59,4 +56,9 @@ YAML (e.g. `defaultCanalFabric16`), update its names and references accordingly
 (e.g. rename to `defaultCanalFabric17`). *DO NOT* alter configuration of preceding
 versions.
 
+### Branch Naming
+For ease of testing, any branch that updates the supported versions of kubernetes
+should have 'test-all' in the branch name.  This will cause gitlab CI to excercise
+all of the supported versions of kubernetes. 
+
```
Comment (guineveresaenger) [inline Documentation/UPGRADING_KUBERNETES.md]: e.g. `git br mypatchfix-test-all`

**Comment by guineveresaenger** [comment]
@coffeepac notice the pipeline didn't run again after your last commit; however since it is a pure documentation change, I will merge regardless.

**Pull request merged**